WordPress/Jetpack/Classes/ViewRelated/WordPress Migration/Types/MigrationFlowCoordinator.swift

```diff
@@ -0,0 +1,72 @@
+import UIKit
+
+final class MigrationFlowCoordinator {
+
+    // MARK: - Properties
+
+    private let account: WPAccount
+
+    private(set) lazy var navigationController: UINavigationController = {
+        let rootViewController = self.viewController(for: currentStep)
+        return MigrationFlowNavigationController(rootViewController: rootViewController)
+    }()
+
+    // MARK: - State
+
+    private var currentStep = MigrationStep.welcome {
+        didSet {
+            self.didUpdateCurrentStep(currentStep)
+        }
+    }
+
+    // MARK: - Init
+
+    init() {
+        let context = ContextManager.sharedInstance().mainContext
+        guard let account = try? WPAccount.lookupDefaultWordPressComAccount(in: context) else {
+            fatalError()
```

Comment:
I totally agree that crashing the app intentionally is bad and I was planning to fix this. The idea I had in mind is exactly like your 2nd suggestion:

>Another option could be to check for the existing account at the call site, and inject the account from there

But I'm not sure I understand this:

>possibly even better, since the account property seems to be only used by the viewModel, then we might as well 
inject the viewModel in the vc, avoiding any dependency on Core Data in the vc.

The `account` is indeed only used by the viewModel, but[ the viewModel is already injected in the VC](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533/files#diff-abe4968c501073aad9d7ce4b9221a6c10e0198052b7e520beb1ced44f1b84fcdR24).

Reply:
I pushed this commit https://github.com/wordpress-mobile/WordPress-iOS/pull/19533/commits/70f066dafe0de0e82bf7e0557a4696367f2d40fa adding `WPAccount` dependency and I updated the temporary code in **Test Instructions > Prerequisites** section.